core/src/main/scala/org/apache/spark/deploy/rest/RestSubmissionServer.scala

```diff
@@ -93,6 +93,7 @@ private[spark] abstract class RestSubmissionServer(
    */
   private def doStart(startPort: Int): (Server, Int) = {
     val threadPool = new QueuedThreadPool(masterConf.get(MASTER_REST_SERVER_MAX_THREADS))
+    threadPool.setName(getClass().getSimpleName())
```

Comment:
- Default value:
<img width="454" alt="image" src="https://github.com/user-attachments/assets/f5b69dee-1945-48ac-b9a1-54b7870f25bd">

- We manually set a value here:
<img width="728" alt="image" src="https://github.com/user-attachments/assets/1605dd1a-6ce0-48e9-bbb3-b1fe1a880be4">